### PR TITLE
Refactor a variant and get rid of boost

### PR DESCRIPTION
**Pull request by havogt**
## Technical Description

Exactly that.

### Resolves / Enhances

Remove a strange usage of boost::variant.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: dawn/src/dawn/IIR/StencilMetaInformation.h [hunk 1]
```diff
@@ -38,6 +38,55 @@ namespace iir {
...
+};
+template <>
+struct GetAccessesOfTypeHelper<FieldAccessType::FAT_Field> {
+  auto const& operator()(FieldAccessMetadata const& fieldAccessMetadata) {
+    return fieldAccessMetadata.FieldAccessIDSet_;
+  }
+};
+template <>
+struct GetAccessesOfTypeHelper<FieldAccessType::FAT_LocalVariable> {
+  void operator()(FieldAccessMetadata const& fieldAccessMetadata) {
```
Comment (lukasm91) [inline dawn/src/dawn/IIR/StencilMetaInformation.h]:
```suggestion
  void operator()(FieldAccessMetadata const&) {
```

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**